### DISCUSSION TITLE
Have Listbox.Options box track the width of the Listbox.Button

### DISCUSSION THREAD

**Discussion by okbytes** [tailwindlabs/headlessui#625] [Help]
I've got a responsive Listbox in a CSS Grid, but I can't quite figure out how to get the Listbox.Options to track the width of the Listbox.Button. I've tried putting a ref on the Listbox.Button and using a LayoutEffect to update the Listbox.Options width, which works for page load and initial component mount/rendering. Any thoughts or ideas are welcome!

Here's what it looks like if the window is shrunk after page load
![image](https://user-images.githubusercontent.com/6867344/122443092-1915e580-cf54-11eb-8713-8071571ab3f9.png)

Here is that same view with page reload
![image](https://user-images.githubusercontent.com/6867344/122443197-30ed6980-cf54-11eb-8d88-8f0a88d31d30.png)

**Comment by calebdeji** [comment]
Hi @okbytes , were you able to get your way around this?

**Reply by okbytes** [reply]
No

**Reply by RobinMalfait** [reply]
The `--button-width` CSS variable is exposed on the `ListboxOptions` component that you can use to constrain the width. You can see an example here:

https://headlessui.com/react/listbox#setting-the-dropdown-width
